qiskit/pulse/instructions/instruction.py

```diff
@@ -0,0 +1,274 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Instruction = Leaf node of schedule.
+"""
+import warnings
+
+from typing import Tuple, List, Iterable, Callable, Optional
+
+from qiskit.pulse.channels import Channel
+from qiskit.pulse.interfaces import ScheduleComponent
+from qiskit.pulse.schedule import Schedule
+from qiskit.pulse.timeslots import Interval, Timeslot, TimeslotCollection
+
+# pylint: disable=missing-return-doc
+
+
+class Instruction(ScheduleComponent):
+    """An abstract class for leaf nodes of schedule."""
+
+    def __init__(self, operands,  # what type??
+                 *channels: List[Channel],
```

Comment:
`Channel` (I learned this recently, https://www.python.org/dev/peps/pep-0484/#arbitrary-argument-lists-and-default-argument-values)

Reply:
Ah, thank you!